File: Win/ui/views/profile_view.py
"""
ProfileView.py — DAS CRM Windows
User Profile & Account Settings
Feature parity with Android ProfileScreen.tsx
"""

from PyQt6.QtWidgets import (
    QWidget, QVBoxLayout, QHBoxLayout, QLabel, QPushButton, QLineEdit,
    QScrollArea, QFrame, QComboBox, QCheckBox, QMessageBox
)
from PyQt6.QtCore import Qt
from PyQt6.QtGui import QFont, QPixmap, QIcon

class ProfileView(QWidget):
    """User Profile and Settings View"""

    def __init__(self, parent=None):
        super().__init__(parent)
        self.setStyleSheet("""
            QWidget { background-color: #090d16; }
            QLabel { color: #f8fafc; }
            QPushButton { color: white; font-weight: bold; border: none; border-radius: 6px; }
            QLineEdit { background-color: #0f172a; color: #ffffff; border: 1px solid #1e293b;
                        border-radius: 6px; padding: 8px; }
            QComboBox { background-color: #0f172a; color: #ffffff; border: 1px solid #1e293b;
                        border-radius: 6px; padding: 6px; }
        """)

        self._build_ui()

    def _build_ui(self):
        """Build profile UI"""
        layout = QVBoxLayout(self)
        layout.setContentsMargins(0, 0, 0, 0)

        scrollArea = QScrollArea()
        scrollArea.setWidgetResizable(True)
        scrollArea.setStyleSheet("QScrollArea { border: none; background-color: #090d16; }")

        scrollWidget = QWidget()
        scrollLayout = QVBoxLayout(scrollWidget)
        scrollLayout.setContentsMargins(16, 16, 16, 24)
        scrollLayout.setSpacing(16)

        # 👤 PROFILE HEADER
        headerCard = self._build_profile_header()
        scrollLayout.addWidget(headerCard)

        # 📋 PERSONAL INFORMATION
        personalCard = self._build_personal_info()
        scrollLayout.addWidget(personalCard)

        # 🔐 ACCOUNT SECURITY
        securityCard = self._build_security_section()
        scrollLayout.addWidget(securityCard)

        # ⚙️ PREFERENCES
        preferencesCard = self._build_preferences()
        scrollLayout.addWidget(preferencesCard)

        # 💾 SAVE BUTTON
        saveLayout = QHBoxLayout()
        saveLayout.addStretch()
        saveBtn = QPushButton("💾 Save Changes")
        saveBtn.setStyleSheet("""
            QPushButton {
                background-color: #10b981;
                padding: 10px 24px;
                border-radius: 8px;
                font-weight: bold;
            }
            QPushButton:hover { background-color: #059669; }
        """)
        saveBtn.clicked.connect(self._save_changes)
        saveLayout.addWidget(saveBtn)
        scrollLayout.addLayout(saveLayout)

        scrollLayout.addStretch()

        scrollArea.setWidget(scrollWidget)
        layout.addWidget(scrollArea)

    def _build_profile_header(self) -> QFrame:
        """Build profile header section"""
        card = QFrame()
        card.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 16px;
                padding: 16px;
            }
        """)

        layout = QHBoxLayout(card)
        layout.setContentsMargins(16, 16, 16, 16)
        layout.setSpacing(16)

        # Avatar placeholder
        avatarLabel = QLabel("👤")
        avatarLabel.setFont(QFont("Segoe UI", 48))
        avatarLabel.setStyleSheet("background-color: #1e293b; border-radius: 12px; width: 80px; height: 80px;")
        layout.addWidget(avatarLabel)

        # User info
        infoLayout = QVBoxLayout()

        nameLabel = QLabel("Mighty Rai")
        nameLabel.setFont(QFont("Segoe UI", 16, QFont.Weight.Bold))
        nameLabel.setStyleSheet("color: #ffffff;")
        infoLayout.addWidget(nameLabel)

        roleLabel = QLabel("👑 Tenant Admin • DAS CRM Enterprise")
        roleLabel.setFont(QFont("Segoe UI", 11))
        roleLabel.setStyleSheet("color: #94a3b8;")
        infoLayout.addWidget(roleLabel)

        emailLabel = QLabel("📧 mighty@dascrm.com")
        emailLabel.setFont(QFont("Segoe UI", 10))
        emailLabel.setStyleSheet("color: #64748b;")
        infoLayout.addWidget(emailLabel)

        infoLayout.addStretch()
        layout.addLayout(infoLayout, 1)

        editBtn = QPushButton("✏️ Edit Photo")
        editBtn.setStyleSheet("""
            QPushButton {
                background-color: #4f46e5;
                padding: 6px 12px;
                border-radius: 6px;
            }
        """)
        layout.addWidget(editBtn)

        return card

    def _build_personal_info(self) -> QFrame:
        """Build personal information section"""
        card = QFrame()
        card.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 16px;
                padding: 12px;
            }
        """)

        layout = QVBoxLayout(card)
        layout.setContentsMargins(12, 12, 12, 12)
        layout.setSpacing(12)

        title = QLabel("📋 Personal Information")
        title.setFont(QFont("Segoe UI", 12, QFont.Weight.Bold))
        title.setStyleSheet("color: #ffffff;")
        layout.addWidget(title)

        # Form fields
        fields = [
            ("First Name", "Mighty"),
            ("Last Name", "Rai"),
            ("Email Address", "mighty@dascrm.com"),
            ("Phone Number", "+91 98765 43210"),
            ("Organization", "DAS CRM Enterprise"),
            ("Job Title", "Tenant Admin"),
        ]

        for label_text, value in fields:
            fieldLayout = QVBoxLayout()
            fieldLayout.setContentsMargins(0, 0, 0, 0)
            fieldLayout.setSpacing(4)

            label = QLabel(label_text)
            label.setFont(QFont("Segoe UI", 10, QFont.Weight.Bold))
            label.setStyleSheet("color: #cbd5e1;")
            fieldLayout.addWidget(label)

            input_field = QLineEdit()
            input_field.setText(value)
            input_field.setMinimumHeight(32)
            fieldLayout.addWidget(input_field)

            layout.addLayout(fieldLayout)

        return card

    def _build_security_section(self) -> QFrame:
        """Build account security section"""
        card = QFrame()
        card.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 16px;
                padding: 12px;
            }
        """)

        layout = QVBoxLayout(card)
        layout.setContentsMargins(12, 12, 12, 12)
        layout.setSpacing(12)

        title = QLabel("🔐 Account Security")
        title.setFont(QFont("Segoe UI", 12, QFont.Weight.Bold))
        title.setStyleSheet("color: #ffffff;")
        layout.addWidget(title)

        # Password section
        passwdLayout = QVBoxLayout()
        passwdLayout.setContentsMargins(0, 0, 0, 0)
        passwdLayout.setSpacing(4)

        passwdLabel = QLabel("Password")
        passwdLabel.setFont(QFont("Segoe UI", 10, QFont.Weight.Bold))
        passwdLabel.setStyleSheet("color: #cbd5e1;")
        passwdLayout.addWidget(passwdLabel)

        passwdLayout.addWidget(QLabel("Last changed: 45 days ago"))

        btnChangePasswd = QPushButton("🔄 Change Password")
        btnChangePasswd.setStyleSheet("""
            QPushButton {
                background-color: #f97316;
                padding: 6px 12px;
                border-radius: 6px;
                width: fit-content;
            }
        """)
        passwdLayout.addWidget(btnChangePasswd)

        layout.addLayout(passwdLayout)

        # 2FA section
        twoFALayout = QHBoxLayout()

        twoFALabel = QLabel("Two-Factor Authentication (2FA)")
        twoFALabel.setFont(QFont("Segoe UI", 10, QFont.Weight.Bold))
        twoFALabel.setStyleSheet("color: #cbd5e1;")
        twoFALayout.addWidget(twoFALabel)
        twoFALayout.addStretch()

        twoFAToggle = QCheckBox("Enabled")
        twoFAToggle.setChecked(True)
        twoFAToggle.setStyleSheet("color: #34d399;")
        twoFALayout.addWidget(twoFAToggle)

        layout.addLayout(twoFALayout)

        # Active sessions
        sessionsLabel = QLabel("🖥️ Active Sessions: 2 devices")
        sessionsLabel.setFont(QFont("Segoe UI", 10))
        sessionsLabel.setStyleSheet("color: #94a3b8;")
        layout.addWidget(sessionsLabel)

        return card

    def _build_preferences(self) -> QFrame:
        """Build preferences section"""
        card = QFrame()
        card.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 16px;
                padding: 12px;
            }
        """)

        layout = QVBoxLayout(card)
        layout.setContentsMargins(12, 12, 12, 12)
        layout.setSpacing(12)

        title = QLabel("⚙️ Preferences")
        title.setFont(QFont("Segoe UI", 12, QFont.Weight.Bold))
        title.setStyleSheet("color: #ffffff;")
        layout.addWidget(title)

        # Theme
        themeLayout = QVBoxLayout()
        themeLabel = QLabel("Theme")
        themeLabel.setFont(QFont("Segoe UI", 10, QFont.Weight.Bold))
        themeLabel.setStyleSheet("color: #cbd5e1;")
        themeLayout.addWidget(themeLabel)

        themeCombo = QComboBox()
        themeCombo.addItems(["🌙 Dark (Default)", "☀️ Light", "🎨 Auto (System)"])
        themeCombo.setCurrentIndex(0)
        themeLayout.addWidget(themeCombo)
        layout.addLayout(themeLayout)

        # Language
        langLayout = QVBoxLayout()
        langLabel = QLabel("Language")
        langLabel.setFont(QFont("Segoe UI", 10, QFont.Weight.Bold))
        langLabel.setStyleSheet("color: #cbd5e1;")
        langLayout.addWidget(langLabel)

        langCombo = QComboBox()
        langCombo.addItems(["🇬🇧 English", "🇮🇳 Hindi", "🇫🇷 French", "🇪🇸 Spanish"])
        langCombo.setCurrentIndex(0)
        langLayout.addWidget(langCombo)
        layout.addLayout(langLayout)

        # Notifications
        notifLayout = QHBoxLayout()
        notifLabel = QLabel("Email Notifications")
        notifLabel.setFont(QFont("Segoe UI", 10, QFont.Weight.Bold))
        notifLabel.setStyleSheet("color: #cbd5e1;")
        notifLayout.addWidget(notifLabel)
        notifLayout.addStretch()

        notifToggle = QCheckBox("Enabled")
        notifToggle.setChecked(True)
        notifToggle.setStyleSheet("color: #34d399;")
        notifLayout.addWidget(notifToggle)
        layout.addLayout(notifLayout)

        return card

    def _save_changes(self):
        """Save profile changes"""
        QMessageBox.information(
            self,
            "✓ Profile Updated",
            "Your profile changes have been saved successfully!"
        )

File: macOs/Sources/DASCRM/Views/Profile/ProfileView.swift
"""
ProfileView.swift — DAS CRM macOS
User Profile & Account Settings
Feature parity with Android ProfileScreen.tsx
"""

import SwiftUI

struct ProfileView: View {
    @State private var firstName = "Mighty"
    @State private var lastName = "Rai"
    @State private var email = "mighty@dascrm.com"
    @State private var phone = "+91 98765 43210"
    @State private var organization = "DAS CRM Enterprise"
    @State private var jobTitle = "Tenant Admin"
    @State private var theme = "Dark"
    @State private var language = "English"
    @State private var emailNotifications = true
    @State private var twoFAEnabled = true
    @State private var showSaveAlert = false

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 16) {
                // 👤 PROFILE HEADER
                HStack(spacing: 16) {
                    Text("👤")
                        .font(.system(size: 48))
                        .frame(width: 80, height: 80)
                        .background(Color(red: 0.12, green: 0.16, blue: 0.23))
                        .cornerRadius(12)

                    VStack(alignment: .leading, spacing: 4) {
                        Text("Mighty Rai")
                            .font(.system(size: 16, weight: .bold))
                            .foregroundColor(.white)
                        Text("👑 Tenant Admin • DAS CRM Enterprise")
                            .font(.system(size: 11))
                            .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                        Text("📧 mighty@dascrm.com")
                            .font(.system(size: 10))
                            .foregroundColor(Color(red: 0.40, green: 0.45, blue: 0.52))
                    }

                    Spacer()

                    Button(action: {}) {
                        Text("✏️ Edit Photo")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(.white)
                            .padding(.horizontal, 12)
                            .padding(.vertical, 6)
                            .background(Color(red: 0.31, green: 0.27, blue: 0.90))
                            .cornerRadius(6)
                    }
                    .buttonStyle(.plain)
                }
                .padding(16)
                .background(Color(red: 0.06, green: 0.09, blue: 0.16))
                .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                .cornerRadius(16)

                // 📋 PERSONAL INFORMATION
                VStack(alignment: .leading, spacing: 12) {
                    Text("📋 Personal Information")
                        .font(.system(size: 12, weight: .bold))
                        .foregroundColor(.white)

                    VStack(spacing: 12) {
                        FormField(label: "First Name", value: $firstName)
                        FormField(label: "Last Name", value: $lastName)
                        FormField(label: "Email Address", value: $email)
                        FormField(label: "Phone Number", value: $phone)
                        FormField(label: "Organization", value: $organization)
                        FormField(label: "Job Title", value: $jobTitle)
                    }
                }
                .padding(12)
                .background(Color(red: 0.06, green: 0.09, blue: 0.16))
                .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                .cornerRadius(16)

                // 🔐 ACCOUNT SECURITY
                VStack(alignment: .leading, spacing: 12) {
                    Text("🔐 Account Security")
                        .font(.system(size: 12, weight: .bold))
                        .foregroundColor(.white)

                    // Password section
                    VStack(alignment: .leading, spacing: 8) {
                        Text("Password")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                        Text("Last changed: 45 days ago")
                            .font(.system(size: 10))
                            .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))

                        Button(action: {}) {
                            Text("🔄 Change Password")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(.white)
                                .padding(.horizontal, 12)
                                .padding(.vertical, 6)
                                .background(Color(red: 0.97, green: 0.45, blue: 0.09))
                                .cornerRadius(6)
                        }
                        .buttonStyle(.plain)
                    }
                    .padding(10)
                    .background(Color(red: 0.02, green: 0.06, blue: 0.12))
                    .cornerRadius(8)

                    // 2FA section
                    HStack {
                        Text("Two-Factor Authentication (2FA)")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                        Spacer()
                        Toggle("", isOn: $twoFAEnabled)
                            .tint(Color(red: 0.2, green: 0.83, blue: 0.60))
                    }
                    .padding(10)
                    .background(Color(red: 0.02, green: 0.06, blue: 0.12))
                    .cornerRadius(8)

                    // Active sessions
                    HStack {
                        Text("🖥️ Active Sessions: 2 devices")
                            .font(.system(size: 10))
                            .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                        Spacer()
                    }
                    .padding(10)
                    .background(Color(red: 0.02, green: 0.06, blue: 0.12))
                    .cornerRadius(8)
                }
                .padding(12)
                .background(Color(red: 0.06, green: 0.09, blue: 0.16))
                .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                .cornerRadius(16)

                // ⚙️ PREFERENCES
                VStack(alignment: .leading, spacing: 12) {
                    Text("⚙️ Preferences")
                        .font(.system(size: 12, weight: .bold))
                        .foregroundColor(.white)

                    VStack(spacing: 12) {
                        // Theme picker
                        VStack(alignment: .leading, spacing: 4) {
                            Text("Theme")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                            Picker("Theme", selection: $theme) {
                                Text("🌙 Dark (Default)").tag("Dark")
                                Text("☀️ Light").tag("Light")
                                Text("🎨 Auto (System)").tag("Auto")
                            }
                            .pickerStyle(.menu)
                            .frame(maxWidth: .infinity, alignment: .leading)
                        }

                        // Language picker
                        VStack(alignment: .leading, spacing: 4) {
                            Text("Language")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                            Picker("Language", selection: $language) {
                                Text("🇬🇧 English").tag("English")
                                Text("🇮🇳 Hindi").tag("Hindi")
                                Text("🇫🇷 French").tag("French")
                                Text("🇪🇸 Spanish").tag("Spanish")
                            }
                            .pickerStyle(.menu)
                            .frame(maxWidth: .infinity, alignment: .leading)
                        }

                        // Notifications toggle
                        HStack {
                            Text("Email Notifications")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                            Spacer()
                            Toggle("", isOn: $emailNotifications)
                                .tint(Color(red: 0.2, green: 0.83, blue: 0.60))
                        }
                    }
                }
                .padding(12)
                .background(Color(red: 0.06, green: 0.09, blue: 0.16))
                .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                .cornerRadius(16)

                // Save button
                HStack {
                    Spacer()
                    Button(action: {
                        showSaveAlert = true
                    }) {
                        Text("💾 Save Changes")
                            .font(.system(size: 11, weight: .bold))
                            .foregroundColor(.white)
                            .padding(.horizontal, 24)
                            .padding(.vertical, 10)
                            .background(Color(red: 0.2, green: 0.83, blue: 0.60))
                            .cornerRadius(8)
                    }
                    .buttonStyle(.plain)
                }

                Spacer()
            }
            .padding(16)
        }
        .background(Color(red: 0.06, green: 0.08, blue: 0.14))
        .navigationTitle("Profile")
        .alert("✓ Profile Updated", isPresented: $showSaveAlert) {
            Button("OK") { }
        } message: {
            Text("Your profile changes have been saved successfully!")
        }
    }
}

struct FormField: View {
    let label: String
    @Binding var value: String

    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text(label)
                .font(.system(size: 10, weight: .bold))
                .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
            TextField("", text: $value)
                .textFieldStyle(.roundedBorder)
                .font(.system(size: 10))
                .padding(.vertical, 4)
        }
    }
}

#Preview {
    NavigationView {
        ProfileView()
    }
}
